conda/common/path.py

```diff
@@ -359,7 +362,10 @@ def _translation(found_path):  # NOQA
 
 
 def which(executable):
-    return find_executable(executable)
+    """Backwards-compatibility wrapper. Use `shutil.which` directly if possible."""
+    from shutil import which
+
+    return which(executable)
```

Comment:
This won't suffice, since the prepending happens after it checks for `path`: https://github.com/python/cpython/blob/c1e71ce56fdb3eab62ad3190d09130f800e54610/Lib/shutil.py#L1494-L1500

`shutil.which` does this to mimic the `which` command line tool I think, which is probably a reasonable assumption, but it does mean a slight change in behavior.

It would be beneficial to find out where `which` is used and in what context to gauge to potential blast radius for this change in behavior. Based on that we could derive the appropriate way to inform the users (deprecation, breaking change etc). This problem might be more academic than real.

Reply:
I see, it does check whether `curdir` is there so we can add it to the end of the list, but again, I concur with you that we might be overthinking this. If anything, I would have found `find_executable`'s behaviour buggy because `curdir` is searched for on Windows almost always.

Reply:
And we can always copy/vendor `find_executable` directly. [It's a small, self-contained function](https://github.com/pypa/setuptools/blob/fca6d742d4a05b7ce5be1328d9eb517ca96cce7e/setuptools/_distutils/spawn.py#L75).

Reply:
Thinking about this more, I think we should be fine.